src/chartjs-plugin-trendline.js

```diff
@@ -45,32 +45,37 @@ const pluginTrendlineLinear = {
     },
 };
 
-addFitter = (datasetMeta, ctx, dataset, xScale, yScale) => {
+const addFitter = (datasetMeta, ctx, dataset, xScale, yScale) => {
     let defaultColor = dataset.borderColor || 'rgba(169,169,169, .6)';
     let colorMin = dataset.trendlineLinear.colorMin || defaultColor;
     let colorMax = dataset.trendlineLinear.colorMax || defaultColor;
     let lineWidth = dataset.trendlineLinear.width || dataset.borderWidth;
     let lineStyle = dataset.trendlineLinear.lineStyle || 'solid';
     let fillColor = dataset.trendlineLinear.fillColor;
 
+    const parsing = typeof datasetMeta.controller.chart.options.parsing === "object" ?
+        datasetMeta.controller.chart.options.parsing : undefined;
+    const xAxisKey = dataset.trendlineLinear.xAxisKey || parsing ? parsing.xAxisKey : "x";
+    const yAxisKey = dataset.trendlineLinear.yAxisKey || parsing ? parsing.yAxisKey : "y";
```

Comment:
Hi @Makanz, I believe this line won't work as intended if `dataset.trendlineLinear.xAxisKey` has value, it'll throw `Cannot read properties of undefined`
these lines should be 👇🏼  

`  const xAxisKey = dataset.trendlineLinear.xAxisKey || (parsing ? parsing.xAxisKey : "x");
   const yAxisKey = dataset.trendlineLinear.yAxisKey || (parsing ? parsing.yAxisKey : "y");`

Reply:
Please make a pull request if you have the possibility. Else I will try to take a look at it in a couple of weeks.

Reply:
@DialaHamadneh I have released a new version were I have made the change.
https://www.npmjs.com/package/chartjs-plugin-trendline/v/2.1.1